unity-environment/Assets/ML-Agents/Examples/Walker/Scripts/WalkerAgent.cs

```diff
@@ -0,0 +1,189 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class WalkerAgent : Agent
+{
+    [Header("Specific to Walker")] 
+    public Vector3 goalDirection;
+    public Transform hips;
+    public Transform chest;
+    public Transform spine;
+    public Transform head;
+    public Transform thighL;
+    public Transform shinL;
+    public Transform footL;
+    public Transform thighR;
+    public Transform shinR;
+    public Transform footR;
+    public Transform armL;
+    public Transform forearmL;
+    public Transform handL;
+    public Transform armR;
+    public Transform forearmR;
+    public Transform handR;
+    public Dictionary<Transform, BodyPart> bodyParts = new Dictionary<Transform, BodyPart>();
+
+    [System.Serializable]
+    public class BodyPart
+    {
+        public ConfigurableJoint joint;
+        public Rigidbody rb;
+        public Vector3 startingPos;
+        public Quaternion startingRot;
+        public WalkerGroundContact groundContactScript;
+    }
+
+    /// <summary>
+    /// Create BodyPart object and add it to dictionary.
+    /// </summary>
+    /// <param name="t"></param>
+    public void SetupBodyPart(Transform t)
+    {
+        BodyPart bp = new BodyPart
+        {
+            rb = t.GetComponent<Rigidbody>(),
+            joint = t.GetComponent<ConfigurableJoint>(),
+            startingPos = t.position,
+            startingRot = t.rotation
+        };
+        bodyParts.Add(t, bp);
+        bp.groundContactScript = t.GetComponent<WalkerGroundContact>();
+    }
+
+    public override void InitializeAgent()
+    {
+        SetupBodyPart(hips);
+        SetupBodyPart(chest);
+        SetupBodyPart(spine);
+        SetupBodyPart(head);
+        SetupBodyPart(thighL);
+        SetupBodyPart(shinL);
+        SetupBodyPart(footL);
+        SetupBodyPart(thighR);
+        SetupBodyPart(shinR);
+        SetupBodyPart(footR);
+        SetupBodyPart(armL);
+        SetupBodyPart(forearmL);
+        SetupBodyPart(handL);
+        SetupBodyPart(armR);
+        SetupBodyPart(forearmR);
+        SetupBodyPart(handR);
+
+    }
+
+    public Quaternion GetJointRotation(ConfigurableJoint joint)
```

Comment:
- This can be a static method. Maybe move to a helper class, if we have one.
- Add method-level description